### PR TITLE
Migration for `HotEvent` for Flutter hot runner

### PR DESCRIPTION
**Pull request by eliasyishak**
Related to tracker issue:
- https://github.com/flutter/flutter/issues/128251

For this migration, the `HotEvent` was being referenced in 2 additional files outside of `lib/src/reporting/events.dart`, shown below in the screenshot

image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/attach.dart [hunk 1]
```diff
@@ -570,6 +572,7 @@ class HotRunnerFactory {
     bool stayResident = true,
     bool ipv6 = false,
     FlutterProject? flutterProject,
+    Analytics? analytics,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/attach.dart:575]: can we make this required? we shouldn't be adding implicit calls to globals.

File: packages/flutter_tools/test/general.shard/run_hot_test.dart [hunk 2]
```diff
@@ -25,6 +26,7 @@ void main() {
       false,
       'test-reason',
       TestUsage(),
+      NoOpAnalytics()
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/run_hot_test.dart:29]: nit
```suggestion
      NoOpAnalytics(),
```

File: packages/flutter_tools/lib/src/commands/attach.dart (hunk 1 above)
Comment (eliasyishak) [reply packages/flutter_tools/lib/src/commands/attach.dart]: Yep, done!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**